File: back-end/src/main/java/br/edu/ifpb/customer2/service/FabricanteService.java
package br.edu.ifpb.customer2.service;

import br.edu.ifpb.customer2.grpc.FabricanteClient;
import br.edu.ifpb.model.Fabricante;
import org.springframework.stereotype.Service;

@Service
public class FabricanteService {


    private FabricanteClient grpc;

    public FabricanteService() {
        this.grpc = new FabricanteClient();
    }

    public Fabricante salvar(Fabricante fabricante) {
        return grpc.salvar(fabricante);
    }

    public Fabricante buscar(int id) {
        return grpc.buscar(id);
    }

    public void deletar(int id) {
        grpc.deletar(id);
    }

    public Fabricante atualizar(Fabricante fabricante) {
        return grpc.atualizar(fabricante);
    }
}

File: frontend/src/app/customer/service/fabricante/fabricante.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Fabricante } from 'src/app/model/fabricante';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class FabricanteService {

  private url: string = environment.API_URL + 'fabricante/';
  constructor(private http: HttpClient) { }

  salvar(fabricante : Fabricante): Observable<HttpResponse<Fabricante>> {
    return this.http.post<Fabricante>( this.url , fabricante, { observe: 'response' });
  }

  atualizar(fabricante : Fabricante): Observable<HttpResponse<Fabricante>> {
    return this.http.put<Fabricante>( this.url , fabricante, { observe: 'response' });
  }

  deletar(id : number): Observable<HttpResponse<Fabricante>> {
    return this.http.delete<any>( this.url + id, { observe: 'response' });
  }

  listarTodos(): Observable<HttpResponse<Fabricante[]>> {
    return this.http.get<Fabricante[]>( this.url, { observe: 'response' });
  }

  listarPorId(id : number): Observable<HttpResponse<Fabricante>> {
    return this.http.get<Fabricante>( this.url + id, { observe: 'response' } );
  }
}
